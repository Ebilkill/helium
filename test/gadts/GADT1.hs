data X a where
    I :: Int -> X Int
    B :: Bool -> X Bool

f :: X a -> a
f (I i) = i
f (B b) = b