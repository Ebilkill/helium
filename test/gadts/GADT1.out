Prelude is up to date
Compiling gadts/GADT1.hs
fromList [(f,X a -> a)]

Length of errors doesn't match

fromList [(f,X a -> a)]
"fromList [(f,X a -> a)]"
(7,4): Type error in pattern application
 pattern          : B b
 constructor      : B
   type           : Bool -> X Bool
   does not match : Int  -> X Int 

Compilation failed with 1 error
